src/DataPaginatorInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App;
+
+use Closure;
+use Yiisoft\Data\Paginator\PaginatorInterface;
+
+interface DataPaginatorInterface extends PaginatorInterface
```

Comment:
I wonder if OffsetPaginator could be used directly with custom data provider and conditions.

Reply:
Theoretically possible, but several methods are missing in OffsetPaginator
But I would recommend allocating an interface for OffsetPaginator

Reply:
Yes, expanding OffsetPaginator is OK if needed.